src/main/java/com/fasterxml/uuid/EthernetAddress.java

```diff
@@ -271,18 +275,81 @@ public static EthernetAddress fromInterface()
             while (en.hasMoreElements()) {
                 NetworkInterface nint = en.nextElement();
                 if (!nint.isLoopback()) {
-                    byte[] data = nint.getHardwareAddress();
-                    if (data != null && data.length == 6) {
-                        return new EthernetAddress(data);
-                    }
+                    return fromInterface(nint);
                 }
             }
         } catch (java.net.SocketException e) {
```

Comment:
Quick note: given that `fromInterface()` handles exception, should this now be removed?

But also, if `fromInterface()` returns null, should that be returned or iteration continued?